cmd/action/mux_test.go

```diff
@@ -153,3 +157,41 @@ func Test_PostgresSchemaDSN(t *testing.T) {
 		})
 	}
 }
+
+func TestMux_OpenAtlas(t *testing.T) {
+	t.Run("MySQL", func(t *testing.T) {
+		for _, u := range []string{
+			"mysql://root:pass@tcp(%s)/",
+			"mysql://root:pass@tcp(%s)/test",
+			"mysql://root:pass@%s",
+			"mysql://root:pass@%s/",
+			"mysql://root:pass@%s/test",
+			"mysql://%s/test",
+		} {
+			calls, l := mockServer(t)
+			require.NoError(t, mysqld.SetLogger(log.New(ioutil.Discard, "", 1)))
+			_, err := action.DefaultMux.OpenAtlas(fmt.Sprintf(u, l.Addr()))
+			require.Error(t, err, "mock server rejects all incoming connections")
+			require.NotZero(t, *calls)
+		}
+	})
+}
+
+func mockServer(t *testing.T) (*int, net.Listener) {
```

Comment:
Didn't really like it, but wanted to add tests to the OpenAtlas part.

There's no way to register a "mock driver" with the same name, because the mysql package already registers it, and `database/sql.Register` will panic in case of multiple calls with the same name.

Another alternative was to call `sql.unregisterAllDrivers`, but this requires using the `unsafe` package which I prefer to avoid.